src/core/containers/HandleLogin/index.js

```diff
@@ -58,7 +59,14 @@ function createLoadData(dispatch) {
             secure: config.get('cookieSecure'),
             maxAge: config.get('cookieMaxAge'),
           });
-          router.push('/search');
+          let to;
+          try {
+            to = browserBase64Decode(state.split(':')[1]);
```

Comment:
I guess the colon separated FxA state was added in another patch but I'd rather see something like a query string. That way we could decode it and fetch data by name. This would also make it easier to shove more data into the state later on.